### PR TITLE
Feat: Implement a soft delete with a bin

### PR DESCRIPTION
**Pull request by EdenComp**
# Description

This PR adds a "soft delete" feature, which allow users to move files to the bin and either restore them or permanently delete them.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it.

# Checklist

- [X] I have assigned this PR to myself
- [X] I have added at least 1 reviewer
- [X] I have added the needed labels
- [X] I have linked this PR to an issue
- [X] I have linked this PR to a milestone
- [X] I have linked this PR to a project
- [ ] I have tested this code
- [ ] I have added / updated tests (unit / functionals / end-to-end / ...)
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

# Additional comments

- Adds a button "Move file/folder to bin" to their respective edition
- Auto deletes files from the bin when the user connects (configurable amount of days)

### CODE REVIEW

File: src/components/DisplayCards.tsx [hunk 1]
```diff
@@ -107,5 +107,20 @@ export const DisplayCards = ({
 			</VStack>
 		);
 	if (index === 4) return <ProfileCard profile={user.contact.contacts[0]} />;
+	if (index === 6)
```
Comment (RezaRahemtola) [inline src/components/DisplayCards.tsx:110]: why 6 and not 5?

It probably has to do with the final return of the config page :eyes:

File: src/components/file/DeleteFile.tsx [hunk 2]
```diff
@@ -58,11 +61,17 @@ const DeleteFile = ({ file, concernedFiles }: DeleteFileProps): JSX.Element => {
 					w="100%"
 					p="0px"
 					mx="4px"
-					onClick={() => onOpen()}
+					onClick={() => {
```
Comment (RezaRahemtola) [inline src/components/file/DeleteFile.tsx:64]: would be better to put this in a dedicated function outside the tags :smile:

File: src/lib/contact.ts [hunk 3]
```diff
@@ -279,6 +279,40 @@ class Contact {
...
+			});
+			if (!fileFound) {
+				const file = sharedFiles.find((f) => f.id === concernedFile.id);
+				if (!file) {
+					return { success: false, message: 'File not found' };
+				}
+				await post.Publish({
+					account: this.account!,
+					postType: 'InterPlanetaryCloud',
+					content: { file: { ...concernedFile, deletedAt }, tags: ['rename', concernedFile.id] },
```
Comment (RezaRahemtola) [inline src/lib/contact.ts:302]: great to support shared content sent to the bin!

But this won't work, you also need to take care of this `rename` case is the `loadUpdates` method

File: src/lib/drive.ts [hunk 4]
```diff
@@ -93,6 +93,15 @@ class Drive {
 		}
 	}
 
+	public async autoDelete() {
+		try {
+			const filesToDelete = this.files.filter((file) => file.deletedAt !== null && Date.now() - file.deletedAt >= 2592000000)
```
Comment (RezaRahemtola) [inline src/lib/drive.ts:98]: this number is really weird, explicit it in a constant with a comment and clean multiplications of hours/days etc

**Review by RezaRahemtola — CHANGES_REQUESTED** [review]

File: src/components/DisplayCards.tsx (hunk 1 above)
Comment (EdenComp) [reply src/components/DisplayCards.tsx]: Fixed, used 5

File: src/lib/contact.ts (hunk 3 above)
Comment (EdenComp) [reply src/lib/contact.ts]: Done

File: src/lib/drive.ts (hunk 4 above)
Comment (EdenComp) [reply src/lib/drive.ts]: Fixed

File: src/components/file/DeleteFile.tsx (hunk 2 above)
Comment (EdenComp) [reply src/components/file/DeleteFile.tsx]: Done

**Review by RezaRahemtola — APPROVED** [review]

**Pull request merged**